### ISSUE TITLE
network code

### ISSUE THREAD

**Issue by renhaofan** [renhaofan/test#1]
# -*- coding: utf-8 -*-

import tensorflow as tf
import numpy as np
import matplotlib.pyplot as plt

def add_layer(inputs,in_size,out_size,activation_function=None):
    Weights = tf.Variable(tf.random_normal([in_size,out_size]))
    biases = tf.Variable(tf.zeros([1,out_size]) + 0.1)#initial biases vaule better not 0, so we add 0.1
    Wx_plus_b = tf.matmul(inputs,Weights) + biases
    if activation_function is None:
        outputs = Wx_plus_b
    else:
        outputs = activation_function(Wx_plus_b)
    return outputs

x_data =np.linspace(-1,1,300)[:,np.newaxis]# square brackets means add demension
noise = np.random.normal(0,0.05,x_data.shape)
y_data =np.square(x_data) - 0.5 + noise

xs = tf.placeholder(tf.float32,[None,1])# 1 because x_data is 1 dimension
ys = tf.placeholder(tf.float32,[None,1])
l1 = add_layer(xs,1,10,activation_function=tf.nn.relu)
prediction = add_layer(l1,10,1,activation_function=None)

loss = tf.reduce_mean(tf.reduce_sum(tf.square(ys - prediction),
                     reduction_indices=[1]))
train_step = tf.train.GradientDescentOptimizer(0.1).minimize(loss)# learning rate almost  < 1

# 旧版本的:init = tf.initialize_all_variables()
init = tf.global_variables_initializer()
sess = tf.Session()
sess.run(init)

fig = plt.figure()
ax = fig.add_subplot(1,1,1)
ax.scatter(x_data,y_data)
plt.ion()
plt.show()# 这里会画出图来，之后就暂停了，所以需要加上 plt.ion

# for i in range(1000):
#     sess.run(train_step,feed_dict={xs:x_data,ys:y_data})# 只要placeholde的值，就要feed_dict
#     if i % 50==0:
#     #    print(sess.run(loss,feed_dict={xs:x_data,ys:y_data}))
#     prediction_value = sess.run(prediction,feed_dict={xs:x_data})
#     lines = ax.plot(x_data,prediction_value,'r-',lw=5)# lw = line width
#     ax.lines.remove(lines[0])# remove 去除上一个线段
#     plt.pause(0.1)

# better as fellows:

for i in range(1000):
    sess.run(train_step,feed_dict={xs:x_data,ys:y_data})# 只要placeholde的值，就要feed_dict
    if i % 50 == 0:
        try:
            ax.lines.remove(lines[0])  # remove 去除上一个线段
        except Exception:
            pass
        prediction_value = sess.run(prediction,feed_dict={xs:x_data})
        lines = ax.plot(x_data,prediction_value,'r-',lw=5)# lw = line width
        plt.pause(0.1)
with tf.Session() as sess:
    # summary_writer = tf.summary.FileWriter("E:\work files\program\py\TFtest\logs", tf.get_default_graph())
    summary_writer = tf.summary.FileWriter("E:\work files\program\py\TFtest\logs", sess.graph)
